ocs_ci/framework/pytest_customization/ocscilib.py

```diff
@@ -363,6 +363,12 @@ def get_cli_param(config, name_of_param, default=None):
         any: value of cli parameter or default value
 
     """
+
+    # Override the default value from environmental variables
+    env_var_name = name_of_param.lstrip("-").replace("-", "_").upper()
+    if env_var_name in os.environ:
+        default = os.environ.get(env_var_name)
```

Comment:
I think we could run into an issue if this value overwrites the `default` set when `get_cli_param` is called. I think we should only set this using env vars if `default` is `None`.

Reply:
Sadly, this didn't work as expected in my testing. It is possible that pytest is resetting the env so the variables are not accepted, here. :-/

Reply:
Hmm, testing locally, I could get `OCP_VERSION` to set the `--ocp-version` argument, I may need to dig a little bit deeper here.

Reply:
is there no way to pass the version other then env and consume via cli from run-ci? consuming via cli is clean.

Reply:
I added some testing output and the option is indeed passed as an environmental variable so that is not a problem.

What happens is that the option gets recognized but it is given a default value from the parser, not the default value that is defined in this function as a parameter. That default value is pretty much always ignored nowadays (as long as the option was defined). This default value from parser can be `None` (and in case of `--ocp-version` it in fact is).

@vasukulkarni The env variables support makes it a lot easier to support the options via a Jenkins job, you just define the argument there and it is passed to `run-ci`, no additional changes are necessary to support it.

Reply:
Couldn't this potentially cause any number of collisions of we're just mapping env vars directly to CLI flags? Most similar mechanisms use a prefix - here's [Terraform's approach](https://www.terraform.io/docs/configuration/variables.html#environment-variables)

Reply:
I added the `OCS_CI_` prefix to the enviromental variables to avoid (most of) the collisions.

I am still not happy about the way the values are fetched via getoption though.

Reply:
What's wrong with `config.getoption`?

Reply:
It works differently than you (would) think. It uses the `default` value only if the option was not _defined_ (as in via `addoption`), there is no need to pass it as a cli argument, i.e. the `default` argument in `get_cli_param` was actually never used, it always returned the default value that was set when the option was added.

There is also another thing that is wrong with it -- you can't really distinguish between the default value and the value set via an argument with it (unless you use the `notset` hack, see my latest commit, here). The `config` object does not give you an option to check for a presence of an argument (well, at least not unless you dig deeper inside its actual implementation).